crates/engineioxide/src/packet.rs

```diff
@@ -200,30 +201,56 @@ impl TryFrom<String> for Packet {
 }
 
 /// An OpenPacket is used to initiate a connection
-#[derive(Debug, Clone, Serialize, PartialEq, PartialOrd)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, PartialOrd)]
 #[serde(rename_all = "camelCase")]
 pub struct OpenPacket {
     sid: Sid,
-    upgrades: Vec<String>,
-    ping_interval: u64,
-    ping_timeout: u64,
+    upgrades: Vec<TransportType>,
```

Comment:
Could you also optimize this by switching to smallvec with a size of 1? This would avoid the vec allocation.